ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorizedInputFormatInterface.java

```diff
@@ -18,11 +18,15 @@
 
 package org.apache.hadoop.hive.ql.exec.vector;
 
+import org.apache.hadoop.hive.conf.HiveConf;
+
+import java.util.Properties;
+
 /**
  * Marker interface to indicate a given input format supports
  * vectorization input.
  */
 public interface VectorizedInputFormatInterface {
 
-  VectorizedSupport.Support[] getSupportedFeatures();
+  VectorizedSupport.Support[] getSupportedFeatures(HiveConf hiveConf, Properties properties);
```

Comment:
To reduce the scope of the required refactoring. We could add this as a new method while keeping the old one too.
```
  VectorizedSupport.Support[] getSupportedFeatures();
   
+ default VectorizedSupport.Support[] getSupportedFeatures(HiveConf hiveConf, Properties properties) {
+   return getSupportedFeatures();
+ }
```

And then in HiveIcebergInputFormat we could have something like:

```
+ public VectorizedSupport.Support[] getSupportedFeatures() {
+   throw new UnsupportedOperationException("This overload of getSupportedFeatures should never be called");
+ }

+ public VectorizedSupport.Support[] getSupportedFeatures(HiveConf hiveConf, Properties properties) {
+   //do the logic you have implemented
+ }
```
```

Reply:
Changed